cmsis-core/cmsis_nvic.h

```diff
@@ -34,6 +34,12 @@
 
 #include "cmsis.h"
 
+/* throw meaningful compiler error if yotta configuration is not provided */
+#if !defined(YOTTA_CFG_CMSIS_NVIC_USER_IRQ_OFFSET)      || !defined(YOTTA_CFG_CMSIS_NVIC_USER_IRQ_NUMBER) || \
```

Comment:
No, that symbol is not defined anywhere. It needs to be:

1. defined to 1 for M3 and above
2. defined to 0 for M0
3. defined by the target for M0P (not sure if it should have a default value in this case, and what value that should be, so I'd assume at this point that the target always needs to define it).

We can probabil define this symbol in `cmsis.h` and let the rest of the system use it as needed (including the link you sent above, which doesn't do the right thing in all cases, as previously discussed).